internal/command/arguments/plan_test.go

```diff
@@ -321,25 +322,27 @@ func TestParsePlan_targetFile(t *testing.T) {
 						End:   hcl.Pos{Line: 1, Column: 11, Byte: 10},
 					},
 				},
-			),
+			},
 		},
 	}
 
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
 
 			targetFileArguments := []string{}
-			wantDiagsExported := tc.wantDiags.ForRPC()
 
 			for _, testFile := range tc.files {
 				testFile.tempFileWriter()
 				defer os.Remove(testFile.filePath)
 				targetFileArguments = append(targetFileArguments, "-target-file="+testFile.filePath)
 
-				if testFile.hasError {
-					wantDiagsExported[0].Source().Subject.Filename = testFile.filePath
+				for _, diag := range tc.wantDiags {
+					if diag.Subject != nil {
```

Comment:
Unsure if I actually need this `if`. My thinking is that if we're iterating over the given diags, they will all have non-nil`Subject` and the Subject will never have a `Filename`.